reco_utils/dataset/criteo_dac.py

```diff
@@ -1,73 +1,164 @@
-    # Make sure a temporal zip file get cleaned up no matter what
-    atexit.register(_clean_up, local_cache_path)
+    download_criteo(size, filename, path)
+    filepath = extract_criteo(size, path, local_cache_path)
+
+    if is_databricks():
+        try:
+            # Driver node's file path
+            tar_datapath = "file:" + filepath
+            ## needs to be on dbfs to load
+            dbutils.fs.cp(tar_datapath, dbfs_datapath, recurse=True)
+            path = dbfs_datapath
+        except:
+            raise ValueError(
+                "To use on a Databricks notebook, dbutils object should be passed as an argument"
+            )
+    else:
+        path = filepath
+
+    schema = _get_spark_schema(header)
+    return spark.read.csv(path, schema=schema, sep="\t", header=False)
+
+
+def download_criteo(size="full", filename="dac.tar.gz", work_directory="."):
+    """Download criteo dataset as a compressed file.
+
+    Args:
+        size (str): Size of criteo dataset. It can be "full" or "sample".
+        filename (str): Filename.
+        work_directory (str): Working directory.
+
+    Returns:
+        str: Path of the downloaded file.
+
+    """
+    url = CRITEO_URL[size]
+    return maybe_download(url, filename, work_directory)
+
+
+def extract_criteo(size, path, compressed_file, remove_after_extraction=True):
+    """Extract Criteo dataset tar.
+
+    Args:
+        size (str): Size of criteo dataset. It can be "full" or "sample".
+        path (str): Path to the file.
+        compressed_file (str): Tar file.
+        remove_after_extraction (bool): Whether or not to remove the tar file after extraction.
     
-    ## download if it doesn't already exist locally
-    maybe_download(
-          "http://labs.criteo.com/wp-content/uploads/2015/04/dac_sample.tar.gz",
-          filename,
-          work_directory=path,
+    Returns:
+        str: Path to the extracted file.
+    """
+    extracted_dir = os.path.join(path, "dac")
+    print(
+        "Extracting component files from {}.".format(compressed_file)
     )
+    with tarfile.open(compressed_file) as tar:
+        tar.extractall(extracted_dir)
 
-    #always extract to a subdirectory of cache_path called dac_sample
-    # extracted_dir=os.path.join(path, "dac_sample")
-    print('Extracting component files from {}.'.format(local_cache_path))
-    with tarfile.open(local_cache_path) as tar:
-        tar.extract('dac_sample.txt', path)
-    
-    _clean_up(local_cache_path)
+    if remove_after_extraction:
+        remove_filepath(compressed_file)
 
-    datapath=os.path.join(path, 'dac_sample.txt')
+    return _manage_data_sizes(size, extracted_dir)
 
-    atexit.register(_clean_up, datapath)
-    
+
+def _manage_data_sizes(size, path):
+    if size == "sample":
+        datapath = os.path.join(path, "dac_sample.txt")
+    elif size == "full":
+        datapath = os.path.join(path, "train.txt")
     return datapath
 
-def _clean_up(filepath):
```

Comment:
I was thinking about clean up the downloaded files after we use it.
There are few issues if we just remove _clean_up because of the behavior of `maybe_download` which is "only download if it is not already downloaded".

1. Downloaded file maybe corrupted (by network issue or user interruption, etc.). Still it will not download since the file is already there.
2. Unit tests will not be able to tell if the data hosting server is down or the url has been changed, etc. (we can make unit-test to remove files after the test)
3. Data is downloaded into the same zip file, `local_cache_path`, by default. If a user downloaded **sample** criteo data and then tries to download **full** data (or in movie-lens, downloaded '100k' and then tries to download '1m'), `maybe_download` will recognize the old cache file so that will skip downloading.

Go-around will be,
Clean up the downloaded zip file after use it but not the unzipped data files; and check if unzipped data file exists and then decide if we want to download or not.

Reply:
As you know, Movielens dataloader is now using `local_cache_path` argument to cache downloaded and extracted files. If None is provided, it uses tempfile which cleans-up for us at the end.

Reply:
yes, I added that code